pymobiledevice3/services/afc.py

```diff
@@ -208,36 +208,43 @@ def __init__(self, lockdown: LockdownClient, service_name='com.apple.afc'):
         super().__init__(lockdown, service_name)
         self.packet_num = 0
 
-    def pull(self, src: str, dst: str, callback=None):
+    def pull(self, src: str, dst: str, callback: Optional[callable] = None):
         """ pull a file/directory from given path into a local file/directory """
-        if callback is not None:
-            callback(src, dst)
-
         src = self.resolve_path(src)
-
         if not self.isdir(src):
             # normal file
-            if os.path.isdir(dst):
-                dst = os.path.join(dst, os.path.basename(src))
-            with open(dst, 'wb') as f:
-                f.write(self.get_file_contents(src))
+            if dst.endswith(posixpath.sep):
+                dst_parent_dir = os.path.realpath(os.path.dirname(dst))
+            else:
+                dst_parent_dir = os.path.dirname(os.path.realpath(dst))
+            if not os.path.isdir(dst_parent_dir):  # check if the parent dir exist
+                raise NotADirectoryError(f'pull: directory {dst} does not exist')
```

Comment:
`raise NotADirectoryError(f'pull: directory {dst_parent_dir} does not exist')`?

Reply:
this is on purpose, act like `cp -r`

Reply:
refactor this area